js/common/modules/@arangodb/graph/graphs-generation.js

```diff
@@ -0,0 +1,184 @@
+/*jshint globalstrict:false, strict:false */
+/*global assertEqual, assertTrue, assertNotEqual, JSON */
+'use strict';
+
+// //////////////////////////////////////////////////////////////////////////////
+// / @brief Pregel Tests: graph generation
+// /
+// / @file
+// /
+// / DISCLAIMER
+// /
+// / Copyright 2022 ArangoDB GmbH, Cologne, Germany
+// /
+// / Licensed under the Apache License, Version 2.0 (the "License")
+// / you may not use this file except in compliance with the License.
+// / You may obtain a copy of the License at
+// /
+// /     http://www.apache.org/licenses/LICENSE-2.0
+// /
+// / Unless required by applicable law or agreed to in writing, software
+// / distributed under the License is distributed on an "AS IS" BASIS,
+// / WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// / See the License for the specific language governing permissions and
+// / limitations under the License.
+// /
+// / Copyright holder is ArangoDB GmbH, Cologne, Germany
+// /
+// / @author Roman Rabinovich
+// //////////////////////////////////////////////////////////////////////////////
+
+const makeEdge = (from, to, vColl, name_prefix) => {
+        return {
+            _from: `${vColl}/${name_prefix}_${from}`,
+            _to: `${vColl}/${name_prefix}_${to}`,
+            vertex: `${name_prefix}_${from}`
+        };
+};
+
+function makeVertex(name, name_prefix) {
+    return {_key: `${name_prefix}_${name}`};
+}
+
+function makeVertices(length, name_prefix) {
+    let vertices = [];
+    for (let i = 0; i < length; ++i) {
+        vertices.push(makeVertex(i, name_prefix));
+    }
+    return vertices;
+}
+
+function makeSingleVertex(name_prefix) {
+    const vertices = makeVertices(1, name_prefix);
+    const edges = [];
+    return {vertices, edges};
+}
+
+// length must be at least 2, shorter cycles make no sense and throw
+// because then the test is wrong
+function makeDirectedCycle(length, vColl, name_prefix) {
```

Comment:
For this and all the following functions:
I like them in general, we should try to reorganize this a bit, such that we can use different vertex / edge generators.
I think it is worthwhile to have the option to switch to a SmartGraph generator, to allow testing the algorithms there as well.

This would only affect replacing / injecting the makeVertex makeEdges methods above.

Reply:
Done